docs/cookbook/named_entity_recognition.md

```diff
@@ -0,0 +1,269 @@
+# Named Entity Recognition
+
+This guide demonstrates techniques to perform Named Entity Recognition (NER) using Large Language Models (LLMs) with various levels of nested entity recognition. We'll use Groq's llama-3.1-8b-instant model, but you can adapt this approach to other models with similar capabilities.
+
+??? tip "Mirascope Concepts Used"
+
+    - [Calls](../learn/calls.md)
+    - [Response Model](../learn/response_models.md)
+    - [Prompt Templates](../learn/prompt_templates.md)
+
+## Background
+
+Named Entity Recognition is a subtask of information extraction that seeks to locate and classify named entities in text into predefined categories such as person names, organizations, locations, etc. LLMs have revolutionized NER by enabling more context-aware and hierarchical entity recognition, going beyond traditional rule-based or statistical methods.
+
+## Setup
+
+First, ensure you have the necessary packages installed and API keys set:
+
+```bash
+pip install mirascope
+```
+
+Set your Groq API key as an environment variable:
+
+```bash
+export GROQ_API_KEY=your_api_key_here
+```
+
+## Implementation
+
+We'll implement NER with different levels of complexity: simple and nested entity recognition. Let's start with the simple version:
+
+### Simple NER
+
+```python
+from mirascope.core import groq, prompt_template
+from pydantic import BaseModel, Field
+from typing import List
+
+class SimpleEntity(BaseModel):
+    entity: str = Field(description="The entity found in the text")
+    label: str = Field(description="The label of the entity (e.g., PERSON, ORGANIZATION, LOCATION)")
+
+class SimpleNER(BaseModel):
```

Comment:
we can drop this and just directly extract `list[SimplyEntity]`

Reply:
To keep the readability, i have followed this format. So that when you use the model in calls like 
```python
@groq.call(
    model="llama-3.1-8b-instant",
    response_model=SimpleNER,
    json_mode=True,
    call_params=groq.GroqCallParams(temperature=0.0),
)
```
In the response_model parameter it will be easy to look at, but can be changed.

Reply:
I have added the changes as requested